mikrotik/provider_test.go

```diff
@@ -25,7 +39,41 @@ func testAccPreCheck(t *testing.T) {
 	if v := os.Getenv("MIKROTIK_USER"); v == "" {
 		t.Fatal("The MIKROTIK_USER environment variable must be set")
 	}
-	if v := os.Getenv("MIKROTIK_PASSWORD"); v == "" {
+	if _, exists := os.LookupEnv("MIKROTIK_PASSWORD"); !exists {
 		t.Fatal("The MIKROTIK_PASSWORD environment variable must be set")
 	}
 }
+
+func testAccDeleteResource(resource *schema.Resource, d *schema.ResourceData, meta interface{}) error {
+	if resource.DeleteContext != nil {
+		var diags diag.Diagnostics
+
+		diags = resource.DeleteContext(context.Background(), d, meta)
+
+		for i := range diags {
+			if diags[i].Severity == diag.Error {
+				return fmt.Errorf("error deleting resource: %s", diags[i].Summary)
+			}
+		}
+
+		return nil
+	}
+
+	return resource.Delete(d, meta)
+}
+
+func testAccCheckResourceDisappears(provider *schema.Provider, resource *schema.Resource, resourceName string) resource.TestCheckFunc {
```

Comment:
Are these functions necessary for this change? This refactor is long overdue, but there is still many resource tests that aren't using it. Can we add this function in this PR, but defer using it until a future one given this PR's size?